Libraries/LibWeb/HTML/HTMLDialogElement.cpp

```diff
@@ -50,18 +50,27 @@ void HTMLDialogElement::removed_from(Node* old_parent, Node& old_root)
 {
     HTMLElement::removed_from(old_parent, old_root);
 
-    // 1. If removedNode's close watcher is not null, then:
-    if (m_close_watcher) {
-        // 1.1. Destroy removedNode's close watcher.
-        m_close_watcher->destroy();
-        // 1.2. Set removedNode's close watcher to null.
-        m_close_watcher = nullptr;
-    }
+    // AD-HOC: When a dialog is removed from the document, we additionally set is_modal to false (below)
+    // and remove it from the open dialogs list (see line 65). This ensures it doesn't interfere with close
+    // requests and matches Chromium's behavior. The other steps below (removing from the top layer and destroying
+    // the close watcher) are mandated by the HTML spec. See https://wpt.live/close-watcher/user-activation/y-dialog-disconnected.html
+    set_is_modal(false);
```

Comment:
If this is not in the spec, should it be? Is there an upstream spec issue that you can reference here?